plugins/search/src/apis.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CatalogApi } from '@backstage/plugin-catalog';
+
+export type Result = {
+  name: string;
+  description: string;
+  owner: string;
+  kind: string;
+  lifecycle: string;
+};
+
+export type SearchResults = Array<Result>;
+
+class SearchApi {
+  private catalogApi: CatalogApi;
+
+  constructor(catalogApi: CatalogApi) {
+    this.catalogApi = catalogApi;
+  }
+
+  private async entities() {
+    const entities = await this.catalogApi.getEntities();
+    return entities.map((result: any) => ({
+      name: result.metadata.name,
+      description: result.metadata.description,
+      owner: result.spec.owner,
+      kind: result.kind,
+      lifecycle: result.spec.lifecycle,
+    }));
+  }
+
+  public getSearchResult(): Promise<SearchResults> {
+    return this.entities();
```

Comment:
Am I right in thinking that future would work pass the filters and query string into this function and perform the filtering here before returning results? I'm thinking about how we would back this off to a backend search engine.

Reply:
yes I hope I can continue to work on the backend search plugin sooner than later, when that's in place I see this api as something that can be responsible for adding filters and similar, sending the query + filters to the backend to get the results -> that's what will be returned here.